.github/workflows/ci.yml

```
@@ -596,14 +602,85 @@ jobs:
+        run: bunx vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
 
-      - name: Deploy to Cloudflare Pages
-        if: github.ref == 'refs/heads/master'
-        uses: cloudflare/wrangler-action@v3
-        with:
-          command: pages deploy docs/build/client --project-name takumi
-          apiToken: ${{ secrets.CF_DOCS_PAGES_TOKEN }}
-          accountId: ${{ secrets.CF_ACCOUNT_ID }}
+      - name: Build Vercel output (preview)
+        if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
+        run: bunx vercel build --target=${{ github.head_ref || github.ref_name }} --token=${{ secrets.VERCEL_TOKEN }}
```

actionlint (1.7.10)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)